### PR TITLE
Golden tests for worker executor serialization

### LINKED ISSUE #811: Golden tests for worker serialization
We need to write golden tests to guarantee we are not breaking worker serialization backward compatibility from 1.0.0.

For this, we have to store binary data serialized with 1.0.0 of various data structures, and have deserialization tests for them.

An initial list of what have to be included:

- every case of the `Oplog` type
- `WorkerStatusRecord` 
- `RedisPromiseState`
- `ScheduledAction`
- Types used in oplog payloads. These should be collected as part of the task, but some of them: 
  - `blob_store::ObjectMetadata`
  - `SerializableError`
  - `SerializableStreamError`
  - `SerializableIpAddresses`
  - `WitValue`

### PR DESCRIPTION
**Pull request by vigoo**
Resolves #811

### CODE REVIEW

File: golem-worker-executor-base/src/durable_host/wasm_rpc/serialized.rs [hunk 1]
```diff
@@ -15,11 +15,27 @@
...
     Completed(Result<WitValue, RpcError>),
 }
+
+impl PartialEq for SerializableInvokeResult {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Failed(e1), Self::Failed(e2)) => e1 == e2,
+            (Self::Pending, Self::Pending) => true,
+            (Self::Completed(Ok(r1)), Self::Completed(Ok(r2))) => {
+                let v1: Value = r1.clone().into();
```
Comment (noise64) [inline golem-worker-executor-base/src/durable_host/wasm_rpc/serialized.rs:33]: is clone unavoidable here? maybe *r1 == *r2 or some other workaround?

File: golem-worker-executor-base/tests/compatibility/v1.rs [hunk 2]
```diff
@@ -0,0 +1,1890 @@
...
+        )
+        .unwrap();
+    let encoded = serialize(&value).unwrap();
+    file.write_all(&encoded).unwrap();
+    file.flush().unwrap();
+}
+
+#[test]
+pub fn worker_status() {
+    let ws1 = WorkerStatus::Running;
```
Comment (noise64) [inline golem-worker-executor-base/tests/compatibility/v1.rs:126]: are there also tests for not introducing new enum cases?

File: golem-worker-executor-base/src/durable_host/wasm_rpc/serialized.rs (hunk 1 above)
Comment (vigoo) [reply golem-worker-executor-base/src/durable_host/wasm_rpc/serialized.rs:33]: No idea did not try to optimize this code :)

File: golem-worker-executor-base/tests/compatibility/v1.rs (hunk 2 above)
Comment (vigoo) [reply golem-worker-executor-base/tests/compatibility/v1.rs:126]: I don't think we need to explicitly deny creating new enum cases. The test covers all the ones that were known for 1.0. So what we are testing is that we have to be able to deserialize all the possible cases created by 1.0. This is covered here. So if you can introduce a new case without breaking this, that's fine (I believe with bincode we can do it if we add the case to the end, but haven't tried).

**Pull request merged**